datadog_lambda/metric.py

```diff
@@ -257,34 +169,39 @@ def decrypt_kms_api_key(kms_client, ciphertext):
     return plaintext
 
 
-# Set API Key
-if not api._api_key:
-    DD_API_KEY_SECRET_ARN = os.environ.get("DD_API_KEY_SECRET_ARN", "")
-    DD_API_KEY_SSM_NAME = os.environ.get("DD_API_KEY_SSM_NAME", "")
-    DD_KMS_API_KEY = os.environ.get("DD_KMS_API_KEY", "")
-    DD_API_KEY = os.environ.get("DD_API_KEY", os.environ.get("DATADOG_API_KEY", ""))
-
-    if DD_API_KEY_SECRET_ARN:
-        api._api_key = boto3.client("secretsmanager").get_secret_value(
-            SecretId=DD_API_KEY_SECRET_ARN
-        )["SecretString"]
-    elif DD_API_KEY_SSM_NAME:
-        api._api_key = boto3.client("ssm").get_parameter(
-            Name=DD_API_KEY_SSM_NAME, WithDecryption=True
-        )["Parameter"]["Value"]
-    elif DD_KMS_API_KEY:
-        kms_client = boto3.client("kms")
-        api._api_key = decrypt_kms_api_key(kms_client, DD_KMS_API_KEY)
-    else:
-        api._api_key = DD_API_KEY
-
-logger.debug("Setting DATADOG_API_KEY of length %d", len(api._api_key))
-
-# Set DATADOG_HOST, to send data to a non-default Datadog datacenter
-api._api_host = os.environ.get(
-    "DATADOG_HOST", "https://api." + os.environ.get("DD_SITE", "datadoghq.com")
-)
-logger.debug("Setting DATADOG_HOST to %s", api._api_host)
+# Set API Key only if extension is not here
+if not should_use_extension:
```

Comment:
Perhaps we should put them into a function? and call it in the wrapper's `__init__` method? I think we can further reduce the scope here, the API key is ONLY needed when `DD_FLUSH_TO_LOG` is `false`.

Reply:
I've updated the condition